### PR TITLE
Db/altitud comp software

### PR DESCRIPTION
**Pull request by danielbernalb**
Added Altitude Compensation for CO2 sensors via software:

CO2correctionAlt() applied  when Altitude is different of zero.

getCO2AltitudeCompensation() obtain the CO2 value from the app.

### CODE REVIEW

File: src/Sensors.cpp [hunk 1]
```diff
@@ -376,6 +378,8 @@ bool Sensors::CO2CM1106Read() {
     if (CO2 > 0) {
         dataReady = true;
         DEBUG("-->[CM1106] read > done!");
+        //alt = 2570;                             // Example test
```
Comment (hpsaturn) [inline src/Sensors.cpp:381]: please remove extra comments

File: src/Sensors.cpp [hunk 2]
```diff
@@ -895,6 +899,24 @@ void Sensors::PMGCJA5Init() {
 void Sensors::dhtInit() {
 }
 
+// Altitude compensation for CO2 sensors without Pressure atm or Altitude compensation
+
+void Sensors::CO2correctionAlt() {
+    //DEBUG("-->[SENSORS] Altitud Compensation for CO2 lectures ON");
+    //Serial.print("-->[SENSORS] CO2 original: ");      // Example test
+    //Serial.print(CO2);                                // Example test
+    //Serial.print(" hpa: ");                           // Example test
```
Comment (hpsaturn) [inline src/Sensors.cpp:908]: please add these lines with DEBUG macro. With this we have this messages when debugging is enable.

File: src/Sensors.cpp [hunk 3]
```diff
@@ -895,6 +899,24 @@ void Sensors::PMGCJA5Init() {
...
+    alt = getCO2AltitudeCompensation();
+    float hpa = 1012 - 0.118 * alt + 0.00000473 * alt * alt;            // Cuadratic regresion formula obtained PA (hpa) from high above the sea
+    //Serial.println(hpa);                              // Example test
```
Comment (hpsaturn) [inline src/Sensors.cpp:911]: same here.

**Review by hpsaturn — CHANGES_REQUESTED** [review]
for now, please make this changes

File: src/Sensors.cpp (hunk 1 above)
Comment (danielbernalb) [reply src/Sensors.cpp:381]: donde

File: src/Sensors.cpp (hunk 2 above)
Comment (danielbernalb) [reply src/Sensors.cpp:908]: done

File: src/Sensors.cpp (hunk 3 above)
Comment (danielbernalb) [reply src/Sensors.cpp:911]: done

File: src/Sensors.cpp [hunk 4]
```diff
@@ -895,6 +897,22 @@ void Sensors::PMGCJA5Init() {
...
+    alt = getCO2AltitudeCompensation();
+    float hpa = 1012 - 0.118 * alt + 0.00000473 * alt * alt;            // Cuadratic regresion formula obtained PA (hpa) from high above the sea
+    DEBUG("-->[SENSORS] Atmospheric pressure calculated in hPa: ", String(hpa).c_str());
+    float CO2cor = (0.016 * ((1013.25 - hpa) /10 ) * (CO2 - 400)) + CO2;       // Increment of 1.6% for every hpa of difference at sea level
+    CO2 = round (CO2cor);
+}
+
+float Sensors::getCO2AltitudeCompensation(){
+    return alt;
+}
```
Comment (hpsaturn) [inline src/Sensors.cpp:914]: this method it is the same than current `getAltitude()` method, it doing the same. I think that maybe only remove this method, and in the CO2correctionAlt() use a new global variable, for example `alt_comp` variable, and also write a new method `setCO2AltitudeCompensation(float altitude)` where you set `alt_comp`. With these changes will be have `alt` for real altitude, for example the current bme680, and alt_comp for sensors that need it.

**Review by hpsaturn — CHANGES_REQUESTED** [review]

File: src/Sensors.cpp (hunk 4 above)
Comment (danielbernalb) [reply src/Sensors.cpp]: Done

File: src/Sensors.hpp [hunk 5]
```diff
@@ -81,6 +81,10 @@ class Sensors {
     // temperature offset (for final temp output)
     float toffset = 0.0;  
 
+    // Altitud compensation variable
+    float alt_comp = 0.0;
+    float altitude = 0.0;
```
Comment (hpsaturn) [inline src/Sensors.hpp:86]: I don't understand why we need `altitude` variable? it is a local variable maybe?

File: src/Sensors.hpp [hunk 6]
```diff
@@ -144,6 +148,7 @@ class Sensors {
     float getGas();
 
     void setTempOffset(float offset);
+    void setCO2AltitudeCompensation(float alt_comp);
```
Comment (hpsaturn) [inline src/Sensors.hpp:151]: please use the same name of parameter used in the implementation, like this:

`void setCO2AltitudeCompensation(float altitude);`

**Review by hpsaturn — CHANGES_REQUESTED** [review]

File: src/Sensors.hpp (hunk 6 above)
Comment (danielbernalb) [reply src/Sensors.hpp]: done

File: src/Sensors.hpp (hunk 5 above)
Comment (danielbernalb) [reply src/Sensors.hpp]: erased

**Pull request merged**